src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/LSPDocumentManagerBase.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    internal abstract class LSPDocumentManagerBase : LSPDocumentManager
```

Comment:
>Base [](start = 46, length = 4)

Nit: I would make the Base the base :) A little confusing that Base inherits from non-base to my small brain. :)

Reply:
Or else I would've made LSPDocumentManager an interface

---
In reply to: [381401352](https://github.com/dotnet/aspnetcore-tooling/pull/1602#discussion_r381401352) [](ancestors = 381401352)

Reply:
Or this could be TrackingLspDocumentManager, since Tracking seems to be the aspect added by these new methods.

Reply:
lol ya I can totally get that. This is a pattern we use for our project system internall where we expose publicly the actual base (`LSPDocumentManager`) and internally we have our own Base that's used to mutate the public type.

Reply:
Will rename `LSPDocumentManagerBase` -> `TrackingLSPDocumentManager`